### PR TITLE
Remove cd configurations

### PR DESCRIPTION
**Pull request by cpgo**

### CODE REVIEW

File: butler/src/app/v2/api/deployments/dto/create-git-request.dto.ts [hunk 1]
```diff
@@ -14,13 +14,19 @@
...
-export interface ISpinnakerConfigurationData {
+import { ApiProperty } from '@nestjs/swagger'
+import { GitProvidersEnum } from '../../../core/configuration/interfaces/git-providers.type'
 
-    gitAccount: string
+export class CreateGitDeploymentDto {
 
-    account: string
+  @ApiProperty()
+  public readonly gitToken: string
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/dto/create-git-request.dto.ts:23]: Change to `token`

File: butler/src/app/v2/api/deployments/dto/create-git-request.dto.ts [hunk 2]
```diff
@@ -14,13 +14,19 @@
...
 
-    namespace: string
+  @ApiProperty()
+  public readonly gitProvider: GitProvidersEnum
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/dto/create-git-request.dto.ts:26]: Change to `provider`

File: butler/src/app/v2/operator/deployments.hook.controller.ts [hunk 3]
```diff
@@ -27,9 +27,9 @@ export class DeploymentsHookController {
   @HttpCode(200)
   @UsePipes(new ValidationPipe({ transform: true }))
   public async finalize(@Body() params: HookParams): Promise<{ status?: unknown, children: [], finalized: boolean, resyncAfterSeconds?: number }> {
-    const deployment = await this.deploymentRepository.findWithComponentsAndConfig(params.parent.spec.deploymentId)
+    // const deployment = await this.deploymentRepository.findWithComponentsAndConfig(params.parent.spec.deploymentId)
```
Comment (lucasbfernandes) [inline butler/src/app/v2/operator/deployments.hook.controller.ts:30]: Remove comment

**Review by lucasbfernandes — CHANGES_REQUESTED** [review]

File: butler/src/tests/v2/integration/use-cases/create-deployment-usecase.integration-spec.ts [hunk 4]
```diff
@@ -60,37 +57,37 @@ describe('CreateDeploymentUsecase v2', () => {
...
-      'workspaceId'
-    )
-    await fixtureUtilsService.createEncryptedConfiguration(cdConfiguration)
+    const encryptedToken = `
+-----BEGIN PGP MESSAGE-----
+
+ww0ECQMCcRYScW+NJZZy0kUBbjTidEUAU0cTcHycJ5Phx74jvSTZ7ZE7hxK9AejbNDe5jDRGbqSd
+BSAwlmwpOpK27k2yXj4g1x2VaF9GGl//Ere+xUY=
+=QGZf
+-----END PGP MESSAGE-----
```
Comment (thallesfreitaszup) [inline butler/src/tests/v2/integration/use-cases/create-deployment-usecase.integration-spec.ts:66]: is this token a generated by moove? if not a think that would be a good test case if we could use a generic token generated by moove to ensure that the the encryption and decryption key are the same

File: butler/src/app/v2/api/deployments/pipes/git-token-decryption.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { Injectable, InternalServerErrorException, PipeTransform } from '@nestjs/common'
+import { CreateDeploymentRequestDto } from '../dto/create-deployment-request.dto'
+import * as openpgp from 'openpgp'
+import { AppConstants } from '../../../core/constants'
+
+@Injectable()
+export class GitTokenDecryptionPipe implements PipeTransform {
+
+  public async transform(deploymentRequest: CreateDeploymentRequestDto): Promise<CreateDeploymentRequestDto> {
+    const encryptedToken = deploymentRequest.git.token
```
Comment (thallesfreitaszup) [inline butler/src/app/v2/api/deployments/pipes/git-token-decryption.pipe.ts:26]: what if another application(orange stack,a pipeline)  wants to integrate with butler? we will assume that they have access to the encryption key? or they will have to pass by moove to deploy?
only  a question

File: butler/src/app/v2/operator/use-cases/reconcile-deployment.usecase.ts [hunk 6]
```diff
@@ -57,11 +54,11 @@ export class ReconcileDeploymentUsecase {
         return { children: specs, resyncAfterSeconds: 5 }
       }
       const previousDeployment = await this.deploymentRepository.findWithComponentsAndConfig(previousDeploymentId)
-      const currentAndPrevious = this.reconcileUseCase.concatWithPrevious(previousDeployment, specs, decryptedConfig)
+      const currentAndPrevious = this.reconcileUseCase.concatWithPrevious(previousDeployment, specs)
       return { children: currentAndPrevious, resyncAfterSeconds: 5 }
     }
 
-    const activeComponents = await this.componentRepository.findActiveComponents(deployment.cdConfiguration.id)
+    const activeComponents = await this.componentRepository.findActiveComponents()
```
Comment (thallesfreitaszup) [inline butler/src/app/v2/operator/use-cases/reconcile-deployment.usecase.ts:61]: shouldn't we ensure that the active components is only about the workspace that is deploying? what if the same component is registered in another namespace/workspace and with the same butler url?

**Review by thallesfreitaszup — APPROVED** [review]
Only questions

**Review by lucasbfernandes — APPROVED** [review]

**Pull request merged**